numba/targets/boxing.py

```diff
@@ -359,6 +386,20 @@ def unbox_array(typ, obj, c):
     return NativeValue(c.builder.load(aryptr), is_error=failed)
 
 
+@unbox(types.SmartArrayType)
+def unbox_smart_array(typ, obj, c):
+    a = cgutils.create_struct_proxy(typ)(c.context, c.builder)
+    arr = c.pyapi.call_method(obj, "host")
+    a.data = c.unbox(typ.as_array, arr).value
```

Comment:
I think that `arr` should be decref'ed after this. Perhaps you want to check refcounts in the tests.